connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookResponseManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.inbound.webhook;
+
+import static io.camunda.connector.impl.Constants.RESPONSE_BODY_EXPRESSION;
+
+import io.camunda.connector.api.inbound.InboundConnectorResult;
+import io.camunda.connector.impl.ConnectorInputException;
+import io.camunda.connector.impl.inbound.InboundConnectorProperties;
+import io.camunda.connector.runtime.core.feel.FeelEngineWrapper;
+import io.camunda.connector.runtime.core.feel.FeelEngineWrapperException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+
+public class WebhookResponseManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(WebhookResponseManager.class);
+
+  public static final String PROCESS_DATA_KEY = "processData";
+
+  private final FeelEngineWrapper feelEngine;
+
+  @Autowired
+  public WebhookResponseManager(final FeelEngineWrapper feelEngine) {
+    this.feelEngine = feelEngine;
+  }
+
+  public ResponseEntity produceResponse(
+      final InboundConnectorResult correlatedResult,
+      final InboundConnectorProperties properties,
```

Comment:
Do we need access to the whole `InboundConnectorProperties` here? Otherwise lets only pass `responseExpression`.

Reply:
While we don't need the whole map of properties here, I would like to encapsulate all response body expression logic here. Because on the other hand, we need to do an extraction of the response body expression in the REST controller. WDYT?

Reply:
But we call this from the REST controller. I explicitly created this object to avoid parsing there :)

Or maybe I didn't get your point? 🤔

Reply:
Also: reworked logic a little bit.

Reply:
You should be able to use the new @FEEL annotation support here I think.

Reply:
> You should be able to use the new @feel annotation support here I think.

I haven't seen `@FEEL` annotation being used for service fields, only in properties records. 
@chillleader could you confirm?

Reply:
> You should be able to use the new FEEL jackson support here I think. You could then simply accept the Function of the expression and evaluate it.

Not sure, I am following, but checking how it's done